testing/integration-tests/subxt-test-macro/src/lib.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright 2019-2023 Parity Technologies (UK) Ltd.
+// This file is dual-licensed as Apache-2.0 or GPL-3.0.
+// see LICENSE for license details.
+
+extern crate proc_macro;
+use proc_macro::TokenStream;
+
+use quote::{format_ident, quote};
+use syn::{
+    parse::{Parse, ParseStream},
+    Error,
+};
+
+/// Environment variable for setting the timeout for the test.
+const SUBXT_TEST_TIMEOUT: &str = "SUBXT_TEST_TIMEOUT";
+
+/// Default timeout for the test.
+const DEFAULT_TIMEOUT: u64 = 60 * 6;
```

Comment:
would be nicer to use Duration here IMO

Reply:
We would still need to pass std::time::Duration to the `quote!` below, which does not impl `toTokens` , I've renamed this to illustrate better that we are counting seconds, thanks!